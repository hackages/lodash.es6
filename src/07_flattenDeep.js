/**
 * Recursively flattens array.
 */
export function flattenDeep(array) {}
