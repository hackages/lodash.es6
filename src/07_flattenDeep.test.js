import {flattenDeep} from './07_flattenDeep'

describe('flattenDeep', () => {
  it('should flatten recursively', () => {
    expect(
      flattenDeep([1, 2, false, {a: 32}, NaN, null, undefined])
    ).toMatchSnapshot()

    expect(
      flattenDeep([[1, 2], [[false, {a: 32}], NaN], [null, [{undefined}]]])
    ).toMatchSnapshot()

    expect(flattenDeep([[[[[[[[[[[[['Sorry']]]]]]]]]]]]])).toMatchSnapshot()
  })
})
