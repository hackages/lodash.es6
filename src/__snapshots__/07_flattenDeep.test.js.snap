// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flattenDeep should flatten recursively 1`] = `
Array [
  1,
  2,
  false,
  Object {
    "a": 32,
  },
  NaN,
  null,
  undefined,
]
`;

exports[`flattenDeep should flatten recursively 2`] = `
Array [
  1,
  2,
  false,
  Object {
    "a": 32,
  },
  NaN,
  null,
  Object {
    "undefined": undefined,
  },
]
`;

exports[`flattenDeep should flatten recursively 3`] = `
Array [
  "Sorry",
]
`;
